src/components/layer-controller/ChannelController.js

```diff
@@ -148,7 +151,35 @@ function ChannelController({
   selectionIndex,
   disableOptions = false,
 }) {
+  const { dtype } = loader;
+  const [domain, setDomain] = useState(null);
   const rgbColor = toRgbUIString(colormapOn, color, theme);
+
+  useEffect(() => {
+    let mounted = true;
+    if (dtype && loader && channels) {
+      const loaderSelection = [
+        { ...channels[channelId].selection },
+      ];
+
+      let domains;
+      getChannelStats({ loader, loaderSelection }).then((stats) => {
+        if (domainType === 'Full') {
+          domains = [[0, DTYPE_VALUES[dtype].max]];
+        } else {
+          domains = stats.map(stat => stat.domain);
+        }
+        const [newDomain] = domains;
+        if (mounted) {
+          setDomain(newDomain);
+        }
+      });
```

Comment:
```suggestion
      let domains;
        if (domainType === 'Full') {
          domains = [[0, DTYPE_VALUES[dtype].max]];
          const [newDomain] = domains;
          if (mounted) {
             setDomain(newDomain);
           }
        } else {
         getChannelStats({ loader, loaderSelection }).then((stats) => {
             domains = stats.map(stat => stat.domain);
             const [newDomain] = domains;
             if (mounted) {
                 setDomain(newDomain);
             }
         });
        }
```

Something like this? I don't think we need to call `getChannelStats` every time, just when we have `Min/Max` as the `domainType`